### DISCUSSION TITLE
使用xmake编译指定target时报error: arm-none-eabi-gcc：error: nosys.specs: No such file or directory

### DISCUSSION THREAD

**Discussion by Jackey990709** [xmake-io/xmake#5441] [Q&A]
我使用xmake编译C工程，有多个target，如图，写在一个xmake.lua中，当使用xmake -- CTSP_LinNm_Set1时，会报error: arm-none-eabi-gcc：error: nosys.specs: No such file or directory的错，xmake -- CTSP_LinNm_Set2却是正常的，这两个target区别只是测试的代码不同。而当我把CTSP_LinNm_Set1这个target单独写到一个xmake.lua中时，运行xmake却又可以正常编译了

**Comment by Jackey990709** [comment]
![image](https://github.com/user-attachments/assets/e0974021-520c-47f0-b4e9-998168132d51)

**Comment by waruqi** [comment]
`-vD` logs 没有，完整配置也没有。编译工具链我也没有。你让我看个啥。。我哪知道 `nosys.specs` 你这怎么配置的，路径对不对。

**Reply by Jackey990709** [reply]
不好意思，公司是内外网隔离的，工程在内网，我是在外网敲了简短的作示例

**Reply by Jackey990709** [reply]
![log](https://github.com/user-attachments/assets/339d41ad-9f94-449f-aad9-872dd68a53f5)
![路径](https://github.com/user-attachments/assets/b3e5a8df-f9a2-4e70-bb20-e2fbfdcd6323)
[log.txt](https://github.com/user-attachments/files/16517115/log.txt)

**Reply by Jackey990709** [reply]
[xmake.lua.txt](https://github.com/user-attachments/files/16517122/xmake.lua.txt)
脚本不支持上传lua文件，我改了一下拓展名

**Reply by waruqi** [reply]
> "-specs=nosys.specs",    
> error: arm-none-eabi-gcc：error: nosys.specs: No such file or directory

这不是提示很明显么，gcc 怎么知道 nosys.specs 文件的路径在哪？你传递是相对路径，gcc 从哪找。。自己改成绝对路径

**Reply by Jackey990709** [reply]
![Feige图片20240807134446](https://github.com/user-attachments/assets/55672025-c5ee-4d5f-9638-8b69b360c646)
用的是绝对路径呀，还是说要改其他地方

**Reply by waruqi** [reply]
你可以自己复制 编译命令 自己试，这好像跟 xmake 没啥关系吧，属于编译器自身的参数使用问题
